python/python/lancedb/table.py

```diff
@@ -127,7 +127,26 @@ def _sanitize_data(
             data, schema = _generator_to_data_and_schema(data)
             if schema is None:
                 raise ValueError("Cannot infer schema from generator data")
-    else:
+    elif isinstance(data, pa.RecordBatch):
+        data = pa.Table.from_batches([data]).to_reader()
+        if schema is None:
+            schema = data.schema
+    elif isinstance(data, pa.dataset.Dataset):
+        data = pa.dataset.Scanner.from_dataset(data).to_reader()
+        if schema is None:
+            schema = data.schema
+    elif isinstance(data, pa.dataset.Scanner):
+        data = data.to_reader()
+        if schema is None:
+            schema = data.schema
+    elif isinstance(data, LanceDataset):
+        data = data.scanner().to_reader()
+        if schema is None:
+            schema = data.schema
+    elif isinstance(data, pa.RecordBatchReader):
+        if schema is None:
+            schema = data.schema
+    elif not isinstance(data, pa.RecordBatchReader):
```

Comment:
The code is ugly, will solve the bugs first before considering reformat the whole _sanitize_data

Reply:
We have similar code path in lance i believe. Can we reuse them>